src/validation.cpp

```diff
@@ -620,19 +621,70 @@ static bool AcceptToMemoryPoolWorker(const CChainParams& chainparams, CTxMemPool
             setAncestors.clear();
```

Comment:
Rather than doing this test after the initial CalculateMemPoolAncestors(), would it make sense to work out if the limits can be relaxed first, and then start calling CalculateMemPoolAncestors? ie

```c++
int bump_desc = 0;
int bump_desc_size = 0;
if (setConflicts.size() == 1) {
 auto conflict = *setIterConflicting.begin();
 if (tx_only_uses_unconfirmed_inputs_from_conflicting_tx) {
  bump_desc = 1;
  bump_desc_size = conflict->GetTxSize();
 }
}
if (!CalculateMemPoolAncestors(..,nLimitDescendents+bump_desc, nLimitDescendantSize+bump_desc_size,..) {
 std::string dummy_err;
 if (nSize >  EXTRA_DESCENDANT_TX_SIZE_LIMIT || !CalculateMemPoolAncestors(..+bump_desc,..+bump_desc_size)) {
  Invalid();
 }
}
```

Reply:
Ah, indeed, nice call, done.